src/AbstractConsumer.php

```diff
@@ -222,18 +225,33 @@ public function shutdown()
     /**
      * Handle exception
      *
+     * Returns true when a message should be requeued; otherwise false.
+     *
      * @param \Throwable $e
-     * @return void
+     * @return bool
      */
     protected function handleException(\Throwable $e)
     {
         if (null === $this->errorCallback) {
-            return;
+            return true;
         }
 
         $callback = $this->errorCallback;
 
-        $callback($e, $this);
+        if (null === $requeue = $callback($e, $this)) {
+            return true;
+        }
+
+        if (! is_bool($requeue)) {
+            $this->logger->error(sprintf(
```

Comment:
When the callback is wrong, we should not log but throw an exception an destroy the php instance. It's not worth keeping it alive when the callback is wrong

Reply:
I agree with you point. Changed.